wikitools/code_block_parser.py

```diff
@@ -0,0 +1,123 @@
+import typing
+
+
+class CodeTag(typing.NamedTuple):
+    start: int
+    len: int
+
+
+class CodeBlock(typing.NamedTuple):
+    """
+    A Markdown code block (monospace font + gray backround). Could be one of the following:
+        - Inline: `text` or `` text `` (the latter is used for snippets containing `)
+        - Multiline: ``` followed by anything else until the next ``` block
+
+    Examples:
+
+        ```test test test```
+        ^ start            ^ end
+                         ^ closed_at
+
+        `this is a very long phrase`
+        ^ start                    ^ end
+                                   ^ closed_at
```

Comment:
why the three properties for this?

Reply:
I made this for convenience, but it seems we never really need `closed_at`